src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,57 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
+
+        try {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+            LocalDate dateFromFormatted = LocalDate.parse(dateFrom, formatter);
+            LocalDate dateToFormatted = LocalDate.parse(dateTo, formatter);
+        } catch (DateTimeParseException exc) {
+            System.out.print("Invalid date format, correct date format is dd.mm.yyyy");
+            throw exc;
+        }
+
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate dateFromFormatted = LocalDate.parse(dateFrom, formatter);
+        LocalDate dateToFormatted = LocalDate.parse(dateTo, formatter);
+
+        StringBuilder builder = new StringBuilder();
+        builder.append("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo);
+
+        if (names != null || data != null) {
```

Comment:
ok, done. I left this validation, but corrected it follow Your suggestion.

Reply:
That's not quite right.

If you want to iterate over both data and names, you need to make sure that BOTH of this variables are not nullable. You can do so by replacing the operator in your former condition, so `if (names != null && data != null)` returns true if both of these objects are not nullable. But since we want to provide an exit condition here and return null immediately if ANY of those two is null, you'll need to check the opposite: `if (names == null || data == null)`.

Please either fix the condition or remove this validation.

Reply:
ok, for this moment I removed validation.